src/libsyntax/parse/parser/item.rs

```diff
@@ -119,14 +119,21 @@ impl<'a> Parser<'a> {
 
             let opt_abi = self.parse_opt_abi()?;
 
+            let constness = if self.eat_keyword(kw::Const) {
```

Comment:
Should be refactored into a method `fn parse_constness(&mut self) -> Constness`.
(Please also search for `kw::Const` to see other places where the keyword is eaten and refactor those also.)